Source/BugsnagClient.m

```diff
@@ -691,35 +695,36 @@ - (void)notify:(NSException *)exception
     //    3 -[BugsnagCrashSentry reportUserException:reason:]
     //    4 -[BugsnagClient notify:message:block:]
 
-    int depth = (int)(BSGNotifierStackFrameCount + report.depth);
+    int depth = (int)(BSGNotifierStackFrameCount + event.depth);
 
-    NSString *reportName =
-        report.errorClass ?: NSStringFromClass([NSException class]);
-    NSString *reportMessage = report.errorMessage ?: @"";
+    NSString *eventErrorClass = event.errorClass ?: NSStringFromClass([NSException class]);
+    NSString *eventMessage = event.errorMessage ?: @"";
 
-    [self.crashSentry reportUserException:reportName
-                                   reason:reportMessage
+    [self.crashSentry reportUserException:eventErrorClass
+                                   reason:eventMessage
                         originalException:exception
                              handledState:[handledState toJson]
                                  appState:[self.state toDictionary]
-                        callbackOverrides:report.overrides
-                                 metadata:[report.metadata copy]
+                        callbackOverrides:event.overrides
+                                 metadata:[event.metadata copy]
                                    config:[self.configuration.config toDictionary]
                              discardDepth:depth];
 
     [self addBreadcrumbWithBlock:^(BugsnagBreadcrumb *_Nonnull crumb) {
-      crumb.type = BSGBreadcrumbTypeError;
-      crumb.message = reportName;
-      crumb.metadata = @{
-          BSGKeyMessage : reportMessage,
-          BSGKeySeverity : BSGFormatSeverity(report.severity)
-      };
+        crumb.type = BSGBreadcrumbTypeError;
+        crumb.message = eventErrorClass;
+        crumb.metadata = @{
+            BSGKeyMessage : eventMessage,
```

Comment:
How would an empty string for `eventMessage` display in the dashboard for an error report? If it would render as an empty string the implementation should be altered to omit the value from the dictionary if it is nil, otherwise no change is required here.

Reply:
![Screenshot 2020-03-27 at 18 36 57](https://user-images.githubusercontent.com/1759027/77788930-05821480-705a-11ea-9e8e-91ddf75d7bc8.png)

Using the example app and hardwiring an empty string gives the attached results, i.e. an empty string is displayed.  `[NSNull null]` resolves to "null" so you're correct, this field needs to be  conditionally added.